br/pkg/task/restore.go

```diff
@@ -633,18 +642,55 @@ func runRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 		}
 	}
 
+	tree, gcID, err := client.InitCheckpoint(ctx, s, &restoreTS, cfg.UseCheckpoint)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	var gcTTL int64 = utils.DefaultBRGCSafePointTTL
+	if cfg.UseCheckpoint {
+		gcTTL = utils.DefaultCheckpointGCSafePointTTL
+	}
 	sp := utils.BRServiceSafePoint{
 		BackupTS: restoreTS,
-		TTL:      utils.DefaultBRGCSafePointTTL,
-		ID:       utils.MakeSafePointID(),
+		TTL:      gcTTL,
+		ID:       gcID,
 	}
 	g.Record("BackupTS", backupMeta.EndVersion)
 	g.Record("RestoreTS", restoreTS)
 
+	cctx, gcSafePointKeeperCancel := context.WithCancel(ctx)
+	gcSafePointKeeperRemovable := false
+	defer func() {
+		// don't reset the gc-safe-point if checkpoint mode is used and restored is not finished
+		if cfg.UseCheckpoint && !gcSafePointKeeperRemovable {
+			log.Info("skip removing gc-safepoint keeper for next retry", zap.String("gc-id", sp.ID))
```

Comment:
Should we keep the PD schedulers pausing too? Or once the region topology get changed, the restoration may cost more time.